src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1269,21 +1272,88 @@ void OrbitMainWindow::ShowTooltip(std::string_view message) {
                      QString::fromUtf8(message.data(), static_cast<int>(message.size())), this);
 }
 
+std::optional<QString> OrbitMainWindow::LoadSourceCode(const std::filesystem::path& file_path) {
```

Comment:
There are several sequential steps in this function: load the file itself, then if it's not successful try to use existing mappings, then if not successful ask user to provide a mapping... I think with comments about these steps it would be easier to read this function

Reply:
I don't like comments too much, but I can split this up into several functions which is actually a good idea. Thanks!

Reply:
Done.